pynuodb/session.py

```diff
@@ -140,8 +153,54 @@ def _split_options(options):
 
         return remote_options, tls_options
 
-    def _open_socket(self, connect_timeout, host, port, read_timeout):
-        self.__sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    @staticmethod
+    def _to_ipaddr(addr):
+        if sys.version_info >= (3, 0):
+            ip = ip_address(str(addr))
+        else:
+            ip = ip_address(unicode(addr,'utf_8'))
+        return ip
+    
+    def _parse_addr(self, addr, options):
+        try:
+            # v4/v6 addr w/o port e.g. 192.168.1.1, 2001:3200:3200::10
+            ip = self._to_ipaddr(addr)
+            port = None
+            ver = ip.version
+        except ValueError:
+            # v4/v6 addr w/port e.g. 192.168.1.1:53, [2001::10]:53
+            parsed = urlparse('//{}'.format(addr))
+            try:
+                ip = self._to_ipaddr(parsed.hostname)                
+                port = parsed.port
+                ver = ip.version
+            except ValueError:
```

Comment:
It looks like this is the "low-tech" parsing attempt after `self._to_ipaddr(addr)` fails in both cases (with and without port). This only works for ipv4?

Reply:
Don't really understand question. The code as written seems to work for all the cases of v4/v6, port/no port. This was best solution I came up with so far without manually parsing every case.  The comments show which form is handled in each case. (I presume you are not looking at any of the intermediate commits; just look at the last)

Reply:
I'm referring to this particular `except`-clause. I can see first we handle "w/o port", then "with port", and then this.

Reply:
The exception at line 177 is hit when we have hostname rather than literal ip addr. (i.e. test@ad0 or test@ad0:47004).

Reply:
I see. Can you add a comment to say that? Thanks.

Reply:
The relevant comments are on lines 180 & 184  ('hostname w/o port...", "hostname with port...").